### ISSUE TITLE
Could this work together with Spring Boot?

### ISSUE THREAD

**Issue by Luthien-in-edhil** [wiverson/maven-jpackage-template#7]
Hi,

this template looks very promising.
I'm eager to try out and see if it works for my [project](https://github.com/aduial/ithildin-jfx). It is a JavaFX desktop app that uses Spring Boot, JPA and a H2 database. I used [FXWeaver](https://github.com/rgielen/javafx-weaver) to make Spring and JavaFX play nice together, and that seems to work perfectly OK, but I am still trying to settle on a way to build packages for Windows, Mac OS and Linux. 

I've worked long enough with Spring (+Boot) to always be a bit suspicious when it comes to any novel use-case ... Spring's invisible magic is great as long as it works for you, but it can also make it extremely hard to figure out when there's an issue.

That's why I wonder if you might already have used them together, or know of another project that did? 
All the best,
Lúthien

**Comment by wiverson** [comment]
I have used Spring Boot for years, and can't think of any reason you couldn't adapt this to get this template to work with your project as described. The template really is just to illustrate how to get Maven to play nicely with JavaFX modules to build stripped JVM with jpackage. The only tricky bit I can think of is some JARs are really best used as modules, not squished into a shaded JAR. That's going to look like module errors, but that's just tweaking build (specifically, exclude those JARs from the shaded JAR).

One of the use scenarios I'm looking at is to create a System Tray app template for folks that want to build a desktop app but the UI is all done via local browser interactions, but that's a bit different from what you are describing.

**Comment by wiverson** [comment]
Also (completely unrelated) Sindarin poet is the coolest thing, Luthien. Just watch out for shiny gems - they can be more trouble than they are worth. ;)

**Comment by Luthien-in-edhil** [comment]
Thank you, that's encouraging to hear, I'll definitely give it a try :)

And those jewels ... alas, that's all too true! 

_I dhoer manath únodui,
trî annon dûr, angren thamas,
am ered gondeb, hithui
a thaur dhúatheb angoeol.
I aearon min hain dorthas,
govanner hai na vedui,
pelanner io anann ennas
úníniel vi daur, linnol._

Long was the way that fate them bore,
O'er stony mountains cold and grey,
Through halls of iron and darkling door,
And woods of nightshade morrowless.
The Sundering Seas between them lay,
And yet at last they met once more,
And long ago they passed away
In the forest, singing sorrowless.

**Comment by Luthien-in-edhil** [comment]
I'm somewhat reluctant to ask because I have a hunch I'm overlooking something silly, but I feel I've checked everything I could think of a couple of times over. I must confess this is the first time I'm using Java modules, and I have a hunch there is something not quite right in that general direction. 
This is what I did so far:

- set up the environment according to the readme.md;
- created a new repo using your template;
- cloned it locally, tried it out, got a TestApp.dmg, installed that and it works fine;
- cloned that locally, merged the code from [my original application](https://github.com/aduial/ithildin-jfx) in that repository;
- changed some settings, like adding the javafx-fxml and javafx-web modules (I should replace the -web one with something more light-weight, I only use it to render rich text);
- added the required Spring boot, H2 and FXWeaver dependencies and the H2 db files  ...

and hit **mvn clean install**

Result:
<img width="311" alt="git 2021-03-04 01-36-31 2" src="https://user-images.githubusercontent.com/446688/109891957-5c409700-7c8a-11eb-8619-948060f61e3f.png">

There's a small .jar with only the classes from the `src/ `directory, and one of around ~30Mb in the `shaded-jar` directory.
The unpacked-shade dir looks as if it contains what I would expect (eg Spring, H2, Hibernate etc.).
But it goes wrong in the **jdeps** step. The java-tool reports: 

`Failed to execute goal io.github.wiverson:jtoolprovider-plugin:1.0.25:java-tool (jdeps) on project ithildin: jdeps 1`
Details:
```
[INFO] 
[INFO] --- jtoolprovider-plugin:1.0.25:java-tool (jdeps) @ ithildin ---
[ERROR] jdeps failed with error code [1]
[ERROR]    --add-modules
[ERROR]    javafx.base,javafx.controls,javafx.graphics,javafx.fxml,javafx.web,java.logging
[ERROR]    --generate-module-info
[ERROR]    /Users/luthien/git/aduial/ithildin-app/target/work
[ERROR]    --module-path
[ERROR]    /Users/luthien/git/aduial/ithildin-app/mac-javafx/javafx-sdk-15.0.1/lib/
[ERROR]    /Users/luthien/git/aduial/ithildin-app/target/shaded-jar/ithildin.jar
[INFO] Error: Missing dependencies: classes not found from the module path and classpath.
To suppress this error, use --ignore-missing-deps to continue.
ithildin
   ch.qos.logback.classic.ViewStatusMessagesServlet   -> javax.servlet.http.HttpServletRequest              not found
   ch.qos.logback.classic.ViewStatusMessagesServlet   -> javax.servlet.http.HttpServletResponse             not found
... (4500 more lines like this)
```

It's a LONG list ;) - it looks like, well, if I would generate the "Effective POM" in IntelliJ and then spell out every method included therein. Again, I have a hunch this is something really silly, but I can't find it  ... 😓
